File: backend/src/main/java/com/voltmart/ecommerce/service/AuthService.java
package com.voltmart.ecommerce.service;

import com.voltmart.ecommerce.dto.auth.AuthRequest;
import com.voltmart.ecommerce.dto.auth.AuthResponse;
import com.voltmart.ecommerce.dto.auth.DeliveryPreferenceRequest;
import com.voltmart.ecommerce.dto.auth.GoogleClientIdResponse;
import com.voltmart.ecommerce.dto.auth.GoogleAuthRequest;
import com.voltmart.ecommerce.dto.auth.PhoneOtpRequest;
import com.voltmart.ecommerce.dto.auth.PhoneOtpRequestResponse;
import com.voltmart.ecommerce.dto.auth.PhoneOtpVerifyRequest;
import com.voltmart.ecommerce.dto.auth.ProfileCompletionRequest;

public interface AuthService {
    AuthResponse adminLogin(AuthRequest request);
    AuthResponse googleLogin(GoogleAuthRequest request);
    PhoneOtpRequestResponse requestPhoneOtp(PhoneOtpRequest request);
    AuthResponse verifyPhoneOtp(PhoneOtpVerifyRequest request);
    GoogleClientIdResponse getGoogleClientId();
    AuthResponse completeProfile(ProfileCompletionRequest request);
    AuthResponse updateDeliveryPreference(DeliveryPreferenceRequest request);
}

File: frontend/src/services/authService.ts
import api from "./api";
import { AuthResponse, DeliveryMode } from "../types/store";

export const adminLogin = async (payload: { email: string; password: string }) => {
  const response = await api.post<AuthResponse>("/auth/admin/login", payload);
  return response.data;
};

export const googleLogin = async (payload: { credential: string }) => {
  const response = await api.post<AuthResponse>("/auth/google", payload);
  return response.data;
};

export const getGoogleClientId = async () => {
  const response = await api.get<{ clientId: string }>("/auth/google/client-id");
  return response.data.clientId;
};

export const requestPhoneOtp = async (payload: { phoneNumber: string }) => {
  const response = await api.post<{
    phoneNumber: string;
    expiresInSeconds: number;
    sent: boolean;
  }>("/auth/otp/request", payload);
  return response.data;
};

export const verifyPhoneOtp = async (payload: { phoneNumber: string; otp: string }) => {
  const response = await api.post<AuthResponse>("/auth/otp/verify", payload);
  return response.data;
};

export const completeProfile = async (payload: {
  fullName: string;
  phoneNumber: string;
  email?: string;
}) => {
  const response = await api.patch<AuthResponse>("/auth/profile", payload);
  return response.data;
};

export const updateDeliveryPreference = async (payload: {
  preferredDeliveryMode: DeliveryMode;
}) => {
  const response = await api.patch<AuthResponse>("/auth/delivery-preference", payload);
  return response.data;
};
